bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/discovery/GreeDeviceFinder.java

```diff
@@ -0,0 +1,179 @@
+
+            // Send the Scan message
+            GreeScanRequestDTO scanGson = new GreeScanRequestDTO();
+            scanGson.t = GREE_CMDT_SCAN;
+            String scanReq = gson.toJson(scanGson);
+            sendData = scanReq.getBytes(StandardCharsets.UTF_8);
+            logger.trace("Sending scan packet to {}", ipAddress.getHostAddress());
+            clientSocket.setSoTimeout(DISCOVERY_TIMEOUT_MS);
+            DatagramPacket sendPacket = new DatagramPacket(sendData, sendData.length, ipAddress, DISCOVERY_TIMEOUT_MS);
+            clientSocket.send(sendPacket);
+
+            // Loop for respnses from devices until we get a timeout.
+            int retries = MAX_SCAN_CYCLES;
+            while ((retries > 0)) {
+                // Receive a response
+                DatagramPacket receivePacket = new DatagramPacket(receiveData, receiveData.length);
+                try {
+                    clientSocket.receive(receivePacket);
+                    InetAddress remoteAddress = receivePacket.getAddress();
+                    int remotePort = receivePacket.getPort();
+
+                    // Read the response
+                    // String modifiedSentence = new String(receivePacket.getData(), StandardCharsets.UTF_8);
+                    // GreeScanResponseDTO scanResponseGson = gson.fromJson(modifiedSentence,
+                    // GreeScanResponseDTO.class);
+                    String modifiedSentence = new String(receivePacket.getData(), StandardCharsets.UTF_8);
+                    StringReader stringReader = new StringReader(modifiedSentence);
+                    GreeScanResponseDTO scanResponseGson = gson.fromJson(new JsonReader(stringReader),
+                            GreeScanResponseDTO.class);
+
+                    // If there was no pack, ignore the response
+                    if (scanResponseGson.pack == null) {
+                        logger.debug("Invalid packet format, ignore");
+                        continue;
+                    }
+
+                    // Decrypt message - a a GreeException is thrown when something went wrong
+                    /*
+                     * scanResponseGson.decryptedPack = GreeCryptoUtil
+                     * .decryptPack(GreeCryptoUtil.getAESGeneralKeyByteArray(), scanResponseGson.pack);
+                     * String decryptedMsg = GreeCryptoUtil.decryptPack(GreeCryptoUtil.getAESGeneralKeyByteArray(),
+                     * scanResponseGson.pack);
+                     */
+                    String decryptedMsg = scanResponseGson.decryptedPack = GreeCryptoUtil
+                            .decryptPack(GreeCryptoUtil.getAESGeneralKeyByteArray(), scanResponseGson.pack);
+                    logger.debug("Response received from address {}: {}", remoteAddress.getHostAddress(), decryptedMsg);
+
+                    // Create the JSON to hold the response values
+                    scanResponseGson.packJson = gson.fromJson(decryptedMsg, GreeScanReponsePackDTO.class);
```

Comment:
You should catch the json syntax exception this throws.

Reply:
done